neqo-transport/src/connection.rs

```diff
@@ -1230,9 +1253,18 @@ impl Connection {
                     false,
                     self.loss_recovery.cwnd_avail(),
                     PNSpace::Initial,
-                    false,
+                    paced,
                 )
             };
```

Comment:
I was thinking about this already before:
I do not like how we calculate recovery timers state, next_delay  and now "do not send because of pacing" in different functions. I would like that someone takes a look at that all.

Here I do not like that we need to return SendOption. Can we maybe extract part starting with "et (pto, cong_avail, min_pn_space, cc_limited) =" ? and combine it with next_delay? (I do not have a concrete solution, I would need o look at the code). This can be salved in a separate issue.

Reply:
Not sure that I completely understand your comment here.

But if it is to point out that process_output now checks timers, then sends a packet (and checks timers as needed in that process), then finally checks timers again, that's a good observation.  That's a good justification for merging process_timer into process_output, because now we can start to simplify.

It seems to me that what you want to do is check all of our timers, and save anything that might affect the sending of packets or the final callback time we might return.  Pacing is a little different in that it doesn't cause an event to be generated; it's enough to call it opportunistically.  (More so as pacing only really applies if you aren't CC limited or sending a PTO.)

That's probably worth doing as a follow-up either way.  It would be good to reduce the number of times we call into the loss recovery code.